Harvest.Web/Middleware/SerilogHelpers.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Serilog;
+using Serilog.Events;
+
+namespace Harvest.Web.Middleware
+{
+    public static class SerilogHelpers
+    {
+        public static LogEventLevel GetLogEventLevel(HttpContext ctx, double _, Exception ex) =>
+            // currently implementing logic from SerilogApplicationBuilderExtensions.DefaultGetLevel
+            ex != null
+                ? LogEventLevel.Error
+                : ctx.Response.StatusCode > 499
+                    ? LogEventLevel.Error
+                    : LogEventLevel.Information;
+
+        public static void EnrichFromRequest(IDiagnosticContext diagnosticContext, HttpContext httpContext)
+        {
+            var request = httpContext.Request;
+
+            // Set all the common properties available for every request
+            diagnosticContext.Set("Host", request.Host);
+            diagnosticContext.Set("Protocol", request.Protocol);
+            diagnosticContext.Set("Scheme", request.Scheme);
```

Comment:
Why sometimes set the context info here and sometimes via the middleware IActionFilter?  Seems to me the action filter could do a fine job setting host/protocol/scheme and other vars just as well as route and action info?

Reply:
Doh! I didn't even notice that both HttpContext and DiagnosticContext are available in EnrichFromRequest.

Reply:
Never mind. I think it has to stay this way.  I don't see any way to get at ActionExecutingContext from the HttpContext.

Reply:
Yea makes sense, sounds like it would be better to move this into an action filter since you can't do it the other way around.  And maybe if we get an action filter we really like that adds all of our regular serilog context stuff we can make a nuget package out of it.  I would love to get this sort of boilerplate out of a single project and into a common lib.  Stuff like all of this info, user info, and client IP/browser info would be great to have either as an enricher, action filter, or just some middleware.